tests/imap/test_folder_sync.py

```diff
@@ -93,7 +95,7 @@ def test_new_uids_synced_when_polling(db, generic_account, inbox_folder,
 
 
 def test_condstore_flags_refresh(db, default_account, all_mail_folder,
-                                 mock_imapclient, monkeypatch):
+                                 mock_imapclient, patched_gevent, monkeypatch):
```

Comment:
Is there a decorator fixture style we could use here? I'll defer to @spang, but I feel like this should be written more like this, since `patched_gevent` wraps the test in setup/teardown rather than providing a fixture?

```
@patched_gevent
def test_condstore_flags_refresh(db, default_account, all_mail_folder...
```

Would also be good to settle on terminology. Not sure if `patched_gevent` / `mock_imapclient` are doing equivalent things? Maybe `mock_gevent`?